CMakeLists.txt

```
@@ -3026,7 +3026,7 @@ if(GTEST_FOUND OR DOWNLOAD_GTEST)
     $<TARGET_OBJECTS:rust-bridge-shared>
     ${DEPS}
   )
-  target_link_libraries(${TARGET_TESTRUNNER} ${MYSQL_LIBRARIES} ${PNG_LIBRARIES} ${GTEST_LIBRARIES} ${LIBS} rust_engine_shared ${TARGET_ANTIBOT})
+  target_link_libraries(${TARGET_TESTRUNNER} ${MYSQL_LIBRARIES} ${PNG_LIBRARIES} ${GTEST_LIBRARIES} ${LIBS} rust_engine_shared ${ENGINE_SHARED} ${TARGET_ANTIBOT})
```

```suggestion
  target_link_libraries(${TARGET_TESTRUNNER} ${PNG_LIBRARIES} ${GTEST_LIBRARIES} ${LIBS_SERVER})
```

To fix the build on Windows. Looks like you are trying to link engine-shared without the libraries used by engine-shared.

---

Cool thanks. But any idea about the runtime error? You can run the unit tests with a debugger on windows right?

---

The tests ran successfully for me after fixing the build. Maybe the error only happens when building with MSVC. MSVC should have support for ASAN (https://learn.microsoft.com/en-us/cpp/sanitizers/asan?view=msvc-170). Alternatively, run the `testrunner.exe` with a debugger.